### PR TITLE
modify NetworkMetrics to use a hashmap to store metrics

### LINKED ISSUE #6704: Get rid of NetworkMetrics
We’re currently using NetworkMetrics structure which holds a `broadcast_messages: IntCounterVec`. This should be replaced with a `pub static BROADCAST_MESSAGES: Lazy<IntCounterVer>` just like we do with all other metrics. This would get rid of the NetworkMetrics structure and do_create_int_counter_vec function while simplifying some of the code and bringing in consistency.

The one complication is that the metric is read through in repeated_announce_accounts test. The way that read happens needs to be changed. Perhaps read directly from the static? Or change GetMetrics message to return a hash map or accept name of the metric and return only that.

### PR DESCRIPTION
**Pull request by abhijeetbhagat**
Fixes #6704

### CODE REVIEW

File: chain/network/src/stats/metrics.rs [hunk 1]
```diff
@@ -156,21 +165,38 @@ impl MessageDropped {
...
+    #[cfg(feature = "test_features")]
+    pub fn get_broadcast_count(&self, msg_type: &'static str) -> u64 {
+        let hm = self.broadcast_messages.lock().unwrap();
+        hm.get(msg_type).map_or(0, |v| *v)
+    }
+
+    #[cfg(not(feature = "test_features"))]
+    pub fn get_broadcast_count(&self, msg_type: &'static str) -> u64 {
+        BROADCAST_MESSAGES.get_metric_with_label_values(&[msg_type]).unwrap().get()
     }
```
[the remark this answers is not in the document]
Comment (abhijeetbhagat) [reply chain/network/src/stats/metrics.rs:201]: But if `test_features` is not enabled, then how do we get the metrics? This is what i am doing in the handler:
```
    fn handle(&mut self, msg: GetBroadcastMessageCount, _ctx: &mut Context<Self>) -> Self::Result {
        self.network_metrics.get_broadcast_count(msg.msg_type)
    }
```

[the remark this answers is not in the document]
Comment (abhijeetbhagat) [reply chain/network/src/stats/metrics.rs:201]: ok, understood. there are a lot of imports in `peer_manager.rs` that are used in our test. shall i also import them only if `test_features` is enabled?

[the remark this answers is not in the document]
Comment (mina86) [reply chain/network/src/stats/metrics.rs:201]: Yes.  In fact CI will fail if there are any unused imports.  If there are a lot of them you might consider putting them inside of the test function.

File: chain/network/src/test_utils.rs [hunk 2]
```diff
@@ -188,16 +187,19 @@ impl Handler<GetInfo> for PeerManagerActor {
...
 
-impl Handler<GetMetrics> for PeerManagerActor {
-    type Result = Arc<NetworkMetrics>;
+impl Handler<GetBroadcastMessageCount> for PeerManagerActor {
+    type Result = u64;
 
-    fn handle(&mut self, _msg: GetMetrics, _ctx: &mut Context<Self>) -> Self::Result {
-        self.network_metrics.clone()
+    fn handle(&mut self, msg: GetBroadcastMessageCount, _ctx: &mut Context<Self>) -> Self::Result {
+        println!("getting {}", msg.msg_type);
```
[the remark this answers is not in the document]
Comment (abhijeetbhagat) [reply chain/network/src/test_utils.rs]: done

File: chain/network/src/stats/metrics.rs [hunk 3]
```diff
@@ -156,21 +165,38 @@ impl MessageDropped {
...
+        let mut map = self.broadcast_messages.lock().unwrap();
+        let count = map.entry(message_name).or_insert(0);
+        *count += 1;
+    }
+
+    #[cfg(not(feature = "test_features"))]
     pub fn inc_broadcast(&self, message_name: &str) {
-        self.broadcast_messages.with_label_values(&[message_name]).inc();
+        BROADCAST_MESSAGES.with_label_values(&[message_name]).inc();
+    }
```
[the remark this answers is not in the document]
Comment (abhijeetbhagat) [reply chain/network/src/stats/metrics.rs]: done

File: chain/network/src/stats/metrics.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhijeetbhagat) [reply chain/network/src/stats/metrics.rs]: done

File: chain/network/src/stats/metrics.rs [hunk 4]
```diff
@@ -156,21 +165,38 @@ impl MessageDropped {
...
         Self {
-            broadcast_messages: do_create_int_counter_vec(
-                "near_broadcast_msg",
-                "Broadcasted messages",
-                &["type"],
-            ),
+            #[cfg(feature = "test_features")]
+            broadcast_messages: Arc::new(Mutex::new(HashMap::new())),
         }
     }
```
[the remark this answers is not in the document]
Comment (abhijeetbhagat) [reply chain/network/src/stats/metrics.rs]: done

File: chain/network/src/stats/metrics.rs [hunk 5]
```diff
@@ -130,6 +134,11 @@ pub static PARTIAL_ENCODED_CHUNK_REQUEST_DELAY: Lazy<Histogram> = Lazy::new(|| {
         .unwrap()
 });
 
+#[cfg(not(feature = "test_features"))]
+pub static BROADCAST_MESSAGES: Lazy<IntCounterVec> = Lazy::new(|| {
```
[the remark this answers is not in the document]
Comment (abhijeetbhagat) [reply chain/network/src/stats/metrics.rs]: done

**Review by mina86 — APPROVED** [review]

**Pull request merged**